src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/query/ProblemMessageServiceTest.java

```diff
@@ -41,9 +47,22 @@ void should_retrieve_an_ready_message() {
         Assertions.assertThat(caseEventMessages.get(0)).isEqualTo("ID:d257fa4f-73ad-4a82-a30e-9acc377f593d:1:1:1-2704");
         Assertions.assertThat(caseEventMessages.get(1)).isEqualTo("ID:ce8467a0-cea9-4a65-99dd-3ae9a94a4453:16:1:1-811");
 
-
         List<CaseEventMessageEntity> messages = caseEventMessageRepository.findByMessageId(caseEventMessages);
         messages.forEach(msg -> Assertions.assertThat(msg.getState()).isEqualTo(MessageState.NEW));
     }
+
+    @Test
+    void should_retrieve_an_ready_message_with_content() {
+        List<CaseEventMessageEntity> messageEntity =
+            caseEventMessageRepository.findByMessageId(singletonList(MESSAGE_ID));
+        String messageContent = messageEntity.get(0).getMessageContent();
+        assertNotNull(messageContent);
+        assertEquals("{\"EventInstanceId\":\"d7ebb30c-8b48-4edf-9e16-f4735b13b214\"}", messageContent);
+        problemMessageService.process(JobName.FIND_PROBLEM_MESSAGES);
+        messageEntity = caseEventMessageRepository.findByMessageId(singletonList(MESSAGE_ID));
+        String messageContentAfterJobRun = messageEntity.get(0).getMessageContent();
+        assertNotNull(messageContentAfterJobRun);
+        assertEquals(messageContent, messageContentAfterJobRun);
+    }
```

Comment:
Does this test fail before your change to use a read-only transaction?  Just want to make sure it reproduces the problem.